xds/internal/balancer/ringhash/ringhash.go

```diff
@@ -247,6 +261,7 @@ func (b *ringhashBalancer) updateAddresses(addrs []resolver.Address) bool {
 			if oldWeight := scInfo.weight; oldWeight != newWeight {
 				scInfo.weight = newWeight
 				b.subConns.Set(addr, scInfo)
+				b.orderedSubConns = append(b.orderedSubConns, scInfo)
```

Comment:
mm, This seems like we only append when there is a weight change for already existing addresses. We would want to append regardless ?

Reply:
Good catch. Fixed.